feat: add `blas/ext/base/sfirst-index-equal` (#13371)
stdlib-js/stdlib: lib/node_modules/@stdlib/blas/ext/base/sfirst-index-equal/docs/repl.txt

    are accessed at runtime.

    Indexing is relative to the first index. To introduce an offset, use a typed
    array view.

    If `N <= 0`, the function returns `-1`.

    Parameters
    ----------
    N: integer
        Number of indexed elements.

    x: Float32Array
        First input array.

    strideX: integer
        Stride length for `x`.

    y: Float32Array
        Second input array.

    strideY: integer
        Stride length for `y`.

    Returns
    -------
    idx: integer
        Index.

    Examples
    --------
    // Standard usage:
    > var x = new {{alias:@stdlib/array/float32}}( [ 1.0, 2.0, 3.0, 4.0 ] );
    > var y = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0 ] );
    > var idx = {{alias}}( x.length, x, 1, y, 1 )
    2

    // Using `N` and stride parameters:
    > x = new {{alias:@stdlib/array/float32}}( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );
    > y = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0, 0.0, 0.0 ] );
    > idx = {{alias}}( 3, x, 2, y, 2 )
    1

    // Using view offsets:
    > var x0 = new {{alias:@stdlib/array/float32}}( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );

>     > var y0 = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0, 0.0, 0.0 ] );

Reviewer: ```suggestion
    > var y0 = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0, 0.0, 0.0, 0.0 ] );
```

    > var x1 = new {{alias:@stdlib/array/float32}}( x0.buffer, x0.BYTES_PER_ELEMENT*1 );
    > var y1 = new {{alias:@stdlib/array/float32}}( y0.buffer, y0.BYTES_PER_ELEMENT*2 );
    > idx = {{alias}}( 3, x1, 2, y1, 2 )
    -1

{{alias}}( N, x, strideX, offsetX, y, strideY, offsetY )
    Returns the index of the first element in a single-precision floating-point
    strided array equal to a corresponding element in another single-precision
    floating-point strided array using alternative indexing semantics.

    While typed array views mandate a view offset based on the underlying

>     buffer, the offset parameter supports indexing semantics based on a

Reviewer: ```suggestion
    buffer, the offset parameters support indexing semantics based on starting
```

>     starting index.

Reviewer: ```suggestion
    indices.
```

    Parameters
    ----------
    N: integer
        Number of indexed elements.

    x: Float32Array
        First input array.

    strideX: integer
        Stride length for `x`.

    offsetX: integer
        Starting index for `x`.

    y: Float32Array
        Second input array.

    strideY: integer
        Stride length for `y`.

    offsetY: integer
        Starting index for `y`.

    Returns
    -------
    idx: integer
        Index.

    Examples
    --------
    // Standard usage:
    > var x = new {{alias:@stdlib/array/float32}}( [ 1.0, 2.0, 3.0, 4.0 ] );
    > var y = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0 ] );
    > var idx = {{alias}}.ndarray( x.length, x, 1, 0, y, 1, 0 )
    2

    // Using offsets:
    > x = new {{alias:@stdlib/array/float32}}( [ 1.0, 2.0, 3.0, 4.0, 5.0, 6.0 ] );

>     > y = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0, 0.0, 0.0 ] );

Reviewer: ```suggestion
    > y = new {{alias:@stdlib/array/float32}}( [ 0.0, 0.0, 3.0, 0.0, 0.0, 0.0, 0.0 ] );
```
